### PR TITLE
Refactor Object & Mesh property handling

### PR DESCRIPTION
**Pull request by neomonkeus**
@niftools/blender-nif-plugin-reviewer 

# Overview
Initial pass to start extracting out some of the mesh code from the nif_import class.

## Detail
- Pulled out Vertex Group & Color import to separate classes under geometry module
- Pulled out more properties from the material classes

- Created an object types class currently: LOD, empties, billboard, future would include things like lights, etc.
- BSShaderCode is cause for alot of mess around material
-- Break out BSShaderProperty import from NiMaterialProperty(the latter will remain broken as it needs to be rewritten anyways)
-- Extract out from material and process separately around texture export

## Fixes Known Issues
[Ordered list of issues fixed by this PR]

# Documentation
Existing docs should cover current functionality

# Testing
Basic testing of happy code paths

## Manual
Build, install, import basic set of nifs from sample library

## Automated
[List of tests run, updated or added to avoid future regressions]

# Additional Information
- Bump dev of pyffi
- Use NifOp to get version info
- TextureWriter, Object are not longer instantiated or reference through chain, several layers up

### CODE REVIEW

File: io_scene_nif/modules/object/object_types/type_import.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+# ***** END LICENSE BLOCK *****
+
+from pyffi.formats.nif import NifFormat
+import bpy
+
+from io_scene_nif.modules.object.object_import import Object
+
+
+class NiTypes:
+
```
Comment (HENDRIX-ZT2) [inline io_scene_nif/modules/object/object_types/type_import.py]: This kinda calls for a singledispatch handler that is called on every ninode-like object

File: io_scene_nif/modules/property/material/material_import.py [hunk 2]
```diff
@@ -162,62 +143,33 @@ def import_material(self, n_mat_prop, n_texture_prop,
...
-        self.nif_import.dict_materials[material_hash] = b_mat
+        self.dict_materials[material_hash] = b_mat
         return b_mat
+
+    def set_material_vertex_mapping(self, b_mesh, f_map, material, n_uvco):
+        if material:
+            # fix up vertex colors depending on whether we had textures in the material
+            mbasetex = self.texturehelper.has_base_texture(material)
+            mglowtex = self.texturehelper.has_glow_texture(material)
+            if b_mesh.vertex_colors:
```
Comment (HENDRIX-ZT2) [inline io_scene_nif/modules/property/material/material_import.py]: pretty sure that all of this is removed from 2.8, could mark as depreciated

File: io_scene_nif/modules/property/shader/shader_import.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+from pyffi.formats.nif import NifFormat
+
+from io_scene_nif.modules.object.object_import import Object
+from io_scene_nif.modules.property.texture.texture_import import Texture
+from io_scene_nif.utility import nif_utils
+from io_scene_nif.utility.util_logging import NifLog
+
+
+class BSShader:
+
```
Comment (HENDRIX-ZT2) [inline io_scene_nif/modules/property/shader/shader_import.py]: So good to see this stuff finally pulled out 😅

File: io_scene_nif/nif_import.py [hunk 4]
```diff
@@ -184,7 +183,8 @@ def import_root(self, root_block):
         #     self.animationhelper.import_text_keys(root_block)
 
         # read the NIF tree
-        self.active_obj_name = ""
+        Object.ACTIVE_OBJ_NAME = ""
```
Comment (HENDRIX-ZT2) [inline io_scene_nif/nif_import.py]: Can't we use bpy.context.active_object if we have to maintain a reference?

File: io_scene_nif/modules/object/object_types/type_import.py (hunk 1 above)
Comment (neomonkeus) [reply io_scene_nif/modules/object/object_types/type_import.py]: Potentially, would need thinking about how to resolve parsing entire tree for armatures

File: io_scene_nif/modules/property/material/material_import.py (hunk 2 above)
Comment (neomonkeus) [reply io_scene_nif/modules/property/material/material_import.py]: Cant wait

File: io_scene_nif/nif_import.py (hunk 4 above)
Comment (neomonkeus) [reply io_scene_nif/nif_import.py]: This is gone in the subsequent PR

**Pull request merged**